src/parser/FStar.Parser.Dep.fs

```diff
@@ -179,12 +180,22 @@ let has_implementation (file_system_map:files_for_module_name) (key:module_name)
     : bool =
     Option.isSome (implementation_of file_system_map key)
 
-(* In public interface *)
-let cache_file_name fn =
-  FStar.Options.prepend_cache_dir
-    (if Options.lax()
-     then fn ^ ".checked.lax"
-     else fn ^ ".checked")
+
+let cache_file_name_internal (fn:string) : string * bool =  //bool indicates if the cache file exists
+  let cache_fn = fn ^ (if Options.lax () then ".checked.lax" else ".checked") in
+  let mname = fn |> module_name_of_file in
+  let should_already_be_there = mname |> Options.should_be_already_cached in
+  let error_msg_does_not_exist = BU.format1 "Expected %s to be already checked but could not find it" mname in
```

Comment:
This formats an error message string unconditionally.
It would be better to do this only if raising an error